github/pkg/adapter/adapter_test.go

```diff
@@ -407,231 +409,108 @@ var testCases = []testCase{
+		t.Run(tc.name, tc.runner(t, server.URL, ce))
 	}
 }
 
+// mockTransport is a simple fake HTTP transport
+type mockTransport func(req *http.Request) (*http.Response, error)
+
+// RoundTrip implements the required RoundTripper interface for
+// mockTransport
+func (mt mockTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	return mt(req)
+}
+
 // runner returns a testing func that can be passed to t.Run.
-func (tc *testCase) runner(t *testing.T, ra Adapter) func(t *testing.T) {
+func (tc *testCase) runner(t *testing.T, url string, ceClient *adaptertest.TestCloudEventsClient) func(t *testing.T) {
 	return func(t *testing.T) {
 		if tc.eventType == "" {
 			t.Fatal("eventType is required for table tests")
 		}
+		body, _ := json.Marshal(tc.payload)
+		req, err := http.NewRequest("POST", url, bytes.NewReader(body))
+		if err != nil {
+			t.Error(err)
+		}
 		eventID := "12345"
 		if tc.eventID != "" {
 			eventID = tc.eventID
 		}
-		hdr := http.Header{}
-		hdr.Set("X-GitHub-Event", tc.eventType)
-		hdr.Set("X-GitHub-Delivery", eventID)
-		evtErr := ra.handleEvent(tc.payload, hdr)
-
-		if err := tc.verifyErr(evtErr); err != nil {
+		req.Header.Set("X-GitHub-Event", tc.eventType)
+		req.Header.Set("X-GitHub-Delivery", eventID)
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
 			t.Error(err)
 		}
-	}
-}
-
-func (tc *testCase) handleRequest(req *http.Request) (*http.Response, error) {
+		defer resp.Body.Close()
 
-	codec := cehttp.Codec{}
-
-	body, err := ioutil.ReadAll(req.Body)
-	if err != nil {
-		return nil, err
-	}
-	msg := &cehttp.Message{
-		Header: req.Header,
-		Body:   body,
-	}
-
-	event, err := codec.Decode(context.Background(), msg)
-	if err != nil {
-		return nil, fmt.Errorf("unexpected error decoding cloudevent: %s", err)
-	}
-
-	if tc.wantCloudEventType != "" && tc.wantCloudEventType != event.Type() {
-		return nil, fmt.Errorf("want cloud event type %s, got %s",
-			tc.wantCloudEventType, event.Type())
-	}
-
-	gotSource := event.Source()
-	if testSource != gotSource {
-		return nil, fmt.Errorf("want source %s, got %s", testSource, gotSource)
-	}
-
-	gotSubject := event.Context.GetSubject()
-	if tc.wantCloudEventSubject != "" && tc.wantCloudEventSubject != gotSubject {
-		return nil, fmt.Errorf("want subject %s, got %s", tc.wantCloudEventSubject, gotSubject)
-	}
-
-	return &http.Response{
-		StatusCode: 200,
-		Body:       ioutil.NopCloser(bytes.NewBufferString("")),
-		Header:     make(http.Header),
-	}, nil
-}
-
-func (tc *testCase) verifyErr(err error) error {
-	wantErr := tc.wantErr || tc.wantErrMsg != ""
-
-	if wantErr && err == nil {
-		return errors.New("want error, got nil")
-	}
-
-	if !wantErr && err != nil {
-		return fmt.Errorf("want no error, got %v", err)
-	}
-
-	if err != nil {
-		if diff := cmp.Diff(tc.wantErrMsg, err.Error()); diff != "" {
-			return fmt.Errorf("incorrect error (-want, +got): %v", diff)
+		if err := tc.validateAcceptedPayload(t, ceClient); err != nil {
```

Comment:
Instead of returning an error here just to call `t.Error` on it, you could simply fail _inside_ the `validateAcceptedPayload` function using `t.Error` or `t.Fatal`. This is a very common pattern in tests.

Reply:
got it